tools/flang1/flang1exe/semutil2.c

```diff
@@ -9857,11 +9880,16 @@ eval_reshape(ACL *arg, DTYPE dtype)
   }
 
   if (orderarg == NULL) {
-    if (src_sz == dest_sz) {
-      return srclist;
-    }
-    for (i = 0; i < rank; i++) {
-      order[i] = i;
+    if (transpose) {
+        order[0] = 1;
+        order[1] = 0;
```

Comment:
Does the semantic analysis for this initializer need to check that the ranks of the input and output are both 2? Or is that check already performed somewhere else?

Reply:
Yes, sema does that check here for the input. https://github.com/flang-compiler/flang/blob/27fd9b2ec77be73c9bd90856056dab33af987c39/tools/flang1/flang1exe/semfunc.c#L7058
The output is not strictly checked probably even for non-constant transpose.

Reply:
Thanks.